northstar/src/runtime/process/init.rs

```diff
@@ -30,11 +30,130 @@ pub(super) struct Init {
     pub seccomp: Option<AllowList>,
 }
 
+/// This is the type used to communicate with the init process from the runtime.
+#[derive(Debug, Serialize, Deserialize)]
+pub enum InitMessage {
+    NewChild { pid: i32 },
+    DeadChild { pid: i32, exit_status: ExitStatus },
+}
+
+/// Init commands
+#[derive(Debug, Serialize, Deserialize)]
+pub enum InitApi {
+    /// Executes an command
+    Exec {
+        path: CString,
+        args: Vec<CString>,
+        env: Vec<CString>,
+    },
+}
+
+/// Init process main function
+fn init_main(
+    mut rt_tx: channel::Sender<InitMessage>,
+    mut rt_rx: channel::Receiver<InitApi>,
+    seccomp: Option<AllowList>,
+) -> ! {
+    enum Payload {
+        ChildExit { pid: i32, exit_status: ExitStatus },
+        Command(InitApi),
+    }
+    let (tx, rx) = mpsc::channel();
+
+    // This thread forwards the commands coming from the runtime
+    let command_tx = tx.clone();
+    let _command_thread = std::thread::spawn(move || loop {
```

Comment:
The `JoinHandle` is not used. The thread will stop when `rx` is closed which can be gracefully done in the main loop. We could join the thread then.

Other wise detach the thread by dropping the handle.

Reply:
Hmm yes this sounds good, like we could add a `InitApi::Shutdown` command to stop init explicitly from the runtime or when the sender is dropped :thinking: